dev-tools/deploy/preflight.py

```
@@ -285,23 +293,140 @@ def is_missing(value: Any) -> bool:
+        or isinstance(exclusive_start, bool)
+        or not isinstance(inclusive_end, int)
+        or isinstance(inclusive_end, bool)
+        or exclusive_start != artifact_sequence
+        or inclusive_end != initial_catchup_sequence
+        or sequence_verification.get("ordered") is not True
+        or any(sequence_verification.get(flag) is not True for flag in required_sequence_flags)
+    ):
+        return (
+            "fail",
+            f"{label} sequenceVerification must prove the canonical bounds and ordered, contiguous, "
+            "complete, gap-free, duplicate-free initial catch-up interval",
```

Ruff (0.16.0)

[warning] Unparenthesized implicit string concatenation in collection

Did you forget a comma?

(ISC004)